src/components/Button/Button.js

```diff
@@ -0,0 +1,204 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import ListPropType from '../../prop-types/list';
+import { ButtonIconPosition, ButtonType, ButtonVariant } from './ButtonEnums';
+import './Button.scss';
+
+const propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    confirmText: PropTypes.string,
+    confirmedText: PropTypes.string,
+    hasConfirm: PropTypes.bool,
+    icon: PropTypes.element,
+    iconPosition: ListPropType([
+        ButtonIconPosition.LEFT,
+        ButtonIconPosition.RIGHT,
+    ]),
+    isActive: PropTypes.bool,
+    isDisabled: PropTypes.bool,
+    isFullWidth: PropTypes.bool,
+    onClick: PropTypes.func.isRequired,
+    style: PropTypes.objectOf(PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+    ])),
+    tabIndex: PropTypes.number,
+    type: ListPropType([
+        ButtonType.BUTTON,
+        ButtonType.RESET,
+        ButtonType.SUBMIT,
+    ]),
+    variant: ListPropType([
+        ButtonVariant.CLEAR,
+        ButtonVariant.DEFAULT,
+        ButtonVariant.PRIMARY,
+        ButtonVariant.ROUND,
+    ]),
+};
+
+const defaultProps = {
+    children: null,
+    className: null,
+    confirmText: 'Confirm?',
+    confirmedText: 'Cool!',
+    hasConfirm: false,
+    icon: null,
+    iconPosition: ButtonIconPosition.LEFT,
+    isActive: false,
+    isDisabled: false,
+    isFullWidth: false,
+    style: null,
+    tabIndex: null,
+    type: ButtonType.BUTTON,
+    variant: ButtonVariant.DEFAULT,
+};
+
+class Button extends Component {
+    state = {
+        confirmed: false,
+        confirming: false,
+    };
+
+    componentWillUnmount() {
+        clearTimeout(this.confirmedTimeout);
+    }
+
+    handleClick = (event) => {
+        const propsOnClick = this.props.onClick;
+
+        if (this.props.isDisabled) {
+            event.preventDefault();
+            return;
+        }
+
+        if (this.props.hasConfirm) {
+            this.setState({
+                confirmed: false,
+                confirming: true,
+            }, () => {
+                this.confirmRef.focus();
+            });
+        } else {
+            propsOnClick(event);
+        }
+    }
+
+    handleRef = (ref) => {
+        this.componentRef = ref;
+    }
+
+    handleConfirmBlur = () => {
+        if (this.state.confirming) {
+            this.setState({
+                confirming: false,
+            });
+        }
+    }
+
+    handleConfirmClick = (event) => {
+        event.stopPropagation();
+        this.setState({
+            confirmed: true,
+            confirming: false,
+        }, () => {
+            // Wait 1s for animation to finish
+            this.confirmedTimeout = setTimeout(() => {
```

Comment:
I think we should keep this for now and then reconsider all our animation options later on...

This animate off via the timeout varies from what we've done with Tooltip

Reply:
Agreed. It would be better to avoid the timeout.